### ISSUE TITLE
Unable to match nested case classes with more than 27 fields total in scala 3

### ISSUE THREAD

**Issue by hodga** [etorreborre/specs2#1207]
Hi, I created a test case that shows the problem:

```
import org.specs2.mutable.Specification

class ATest extends Specification {

  case class B(
      a23: String,
      a24: String,
      a25: String,
      a26: String,
      a27: String,
      a28: String
  )
  case class A(
      a1: String,
      a2: String,
      a3: String,
      a4: String,
      a5: String,
      a6: String,
      a7: String,
      a8: String,
      a9: String,
      a10: String,
      a11: String,
      a12: String,
      a13: String,
      a14: String,
      a15: String,
      a16: String,
      a17: String,
      a18: String,
      a19: String,
      a20: String,
      a21: String,
      a22: B
  )

  "case class A" should {

    "be equal to itself" in {

      val msg1 =
        A("", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", B("", "", "", "", "", ""))

      msg1 must beEqualTo(msg1)

    }
  }
}
```

It returns this error:

```
[error]    |      msg1 must beEqualTo(msg2)
[error]    |                               ^
[error]    |           method summonAll is declared as `inline`, but was not inlined
[error]    |
[error]    |           Try increasing `-Xmax-inlines` above 32
[error]    |----------------------------------------------------------------------------
[error]    |Inline stack trace
[error]    |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
[error]    |This location contains code that was inlined from Diffable.scala:81
[error]    |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
[error]    |This location contains code that was inlined from Diffable.scala:81
[error]    |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
[error]    |This location contains code that was inlined from Diffable.scala:81
[error]    |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
[error]    |This location contains code that was inlined from Diffable.scala:81
[error]    |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
[error]    |This location contains code that was inlined from Diffable.scala:81
[error]    |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
[error]    |This location contains code that was inlined from Diffable.scala:81
[error]    |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
[error]    |This location contains code that was inlined from Diffable.scala:81
[error]    |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
[error]    |This location contains code that was inlined from Diffable.scala:81
[error]    |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
[error]    |This location contains code that was inlined from Diffable.scala:81
[error]    |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
[error]    |This location contains code that was inlined from Diffable.scala:81
[error]    |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
[error]    |This location contains code that was inlined from Diffable.scala:81
[error]    |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
[error]    |This location contains code that was inlined from Diffable.scala:81
[error]    |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
[error]    |This location contains code that was inlined from Diffable.scala:81
[error]    |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
[error]    |This location contains code that was inlined from Diffable.scala:81
[error]    |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
[error]    |This location contains code that was inlined from Diffable.scala:81
[error]    |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
[error]    |This location contains code that was inlined from Diffable.scala:81
[error]    |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
[error]    |This location contains code that was inlined from Diffable.scala:81
[error]    |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
[error]    |This location contains code that was inlined from Diffable.scala:81
[error]    |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
[error]    |This location contains code that was inlined from Diffable.scala:81
[error]    |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
[error]    |This location contains code that was inlined from Diffable.scala:81
[error]    |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
[error]    |This location contains code that was inlined from Diffable.scala:81
[error]    |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
[error]    |This location contains code that was inlined from Diffable.scala:81
[error]    |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
[error]    |This location contains code that was inlined from Diffable.scala:81
[error]    |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
[error]    |This location contains code that was inlined from Diffable.scala:81
[error]     ----------------------------------------------------------------------------
```

I am guessing that is this line:
https://github.com/etorreborre/specs2/blob/main/matcher/shared/src/main/scala/org/specs2/matcher/describe/Diffable.scala#L81C31-L81C31

Tested with 
* scala 3.2.2 and specs2 5.2.0
* scala 3.3.1 and specs2 5.4.0 

Test passes if you:
* Remove 1 parameter in any of the case classes.
* Put 28 fields in a single case class.
 
Unfortunately I cannot rely on the -Xmax-inlines argument because I need this for a shared data model lib, so the error would just show up in tests in depending projects.

If it helps this is the only similar issue I could find, in Circe: https://github.com/circe/circe/issues/1980

Thank you for making Specs2 :)

**Comment by etorreborre** [comment]
That's an interesting one :-). I think I have a working fix for it, stay tuned.

**Comment by etorreborre** [comment]
The issue is now fixed and will be available in `specs2-5.4.1` shortly.

**Comment by hodga** [comment]
Thank you, this is great! I can confirm it works in my code as well :)
